ecommerce/management/commands/generate_discount_code.py

```diff
@@ -0,0 +1,175 @@
+"""
+Generates discount (sometimes called enrollment) codes with some parameters. 
+This is meant to be an easy way to make these in bulk - for one-offs, you can 
+use this or go through Django Admin (or, eventually, the Staff Dashboard). 
+"""
+
+from django.core.management import BaseCommand
+
+import csv
+import uuid
+import dateutil
+import pytz
+
+from ecommerce.models import Discount
+from ecommerce.constants import (
+    REDEMPTION_TYPE_ONE_TIME,
+    REDEMPTION_TYPE_UNLIMITED,
+    ALL_DISCOUNT_TYPES,
+)
+from users.models import User
+from main.settings import TIME_ZONE
+from decimal import Decimal
+
+
+class Command(BaseCommand):
+    """
+    Generates discount (sometimes called enrollment) codes with some parameters.
+    """
+
+    help = "Generates discount/enrollment codes."
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--prefix",
+            type=str,
+            help="The prefix to use for the codes. (Maximum length 13 characters)",
+        )
+
+        parser.add_argument(
+            "--expires",
+            type=str,
+            help="Optional expiration date for the code, in ISO-8601 (YYYY-MM-DD) format.",
+        )
+
+        parser.add_argument(
+            "--discount-type",
+            type=str,
+            help="Sets the discount type (dollars-off, percent-off, fixed-price; default percent-off)",
+            default="percent-off",
+        )
+
+        parser.add_argument(
+            "--amount",
+            type=str,
+            nargs="?",
+            help="Sets the discount amount",
+            required=True,
+        )
+
+        parser.add_argument(
+            "--count",
+            type=int,
+            nargs="?",
+            help="Number of codes to produce",
+            default=1,
+        )
+
+        parser.add_argument(
+            "--one-time",
+            help="Make the resulting code(s) one-time redemptions (otherwise, default to unlimited)",
+            nargs="?",
+            const=True,
+            default=False,
+        )
+
+        parser.add_argument(
+            "codes",
```

Comment:
IMO we should add a bit more detail about using codes. Maybe in the module docs or here about using codes.